src/components/structures/RoomView.js

```diff
@@ -360,6 +360,10 @@ module.exports = React.createClass({
                 });
             }
         }
+
+        // update ther tab complete list as it depends on who most recently spoke,
+        // and that has probably just changed
```

Comment:
I'm worried about the possible performance implications of this. For *every* new event which appears in the room, we will do `num_timeline_events` operations to populate the `lastSpoke` dictionary. This will then be followed by an NlogN sorting operation on the member list. For rooms like Matrix HQ (or basically any large IRC bridged room), this could prove costly.

A better way to sort them would be to sort them *once* on `componentDidMount`, and then as new messages come in, just take that one element (the user who sent the event) and bump it to the top of the list. Is it a preemie? Probably. But I suspect it might be one we want to make.